src/EditorFeatures/CSharpTest/CodeActions/InitializeParameter/AddParameterCheckTests.cs

```diff
@@ -3582,4 +3582,348 @@ void M(object o, DayOfWeek dayOfWeek, string s)
+                
+                class C
+                {
+                    void M({{validNumericType}} [|num|])
+                    {
+                        if (num < 0)
+                        {
+                            throw new ArgumentOutOfRangeException(nameof(num), num, $"{{string.Format(FeaturesResources._0_cannot_be_negative, "{nameof(num)}").Replace("\"", "\\\"")}}");
+                        }
+                    }
+                }
+                """,
+            ReferenceAssemblies = ReferenceAssemblies.NetStandard.NetStandard20,
+            CodeActionIndex = 0,
+        }.RunAsync();
+
+        await new VerifyCS.Test()
+        {
+            TestCode = code,
+            FixedCode = $$"""
+                using System;
+                
+                class C
+                {
+                    void M({{validNumericType}} [|num|])
+                    {
+                        if (num <= 0)
+                        {
+                            throw new ArgumentOutOfRangeException(nameof(num), num, $"{{string.Format(FeaturesResources._0_cannot_be_negative_or_zero, "{nameof(num)}").Replace("\"", "\\\"")}}");
+                        }
+                    }
+                }
+                """,
+            ReferenceAssemblies = ReferenceAssemblies.NetStandard.NetStandard20,
+            CodeActionIndex = 1,
+        }.RunAsync();
+    }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/37653")]
+    [InlineData("byte")]
+    [InlineData("ushort")]
+    [InlineData("uint")]
+    [InlineData("ulong")]
+    [InlineData("float")]
+    [InlineData("double")]
+    public async Task TestNoNumericChecksForUnsignedAndFloatingPointNumericTypes(string invalidNumericType)
+    {
+        await VerifyCS.VerifyRefactoringAsync($$"""
+            using System;
+            
+            class C
+            {
+                void M({{invalidNumericType}} [|num|])
+                {
+                }
+            }
+            """);
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/37653")]
+    public async Task TestNoNumericChecksForOutParameter()
+    {
+        await VerifyCS.VerifyRefactoringAsync("""
+            using System;
+            
+            class C
+            {
+                void M(out int [|num|])
+                {
+                    num = 0;
+                }
+            }
+            """);
+    }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/37653")]
+    [InlineData("ThrowIfNegative(num)")]
+    [InlineData("ThrowIfNegativeOrZero(num)")]
+    [InlineData("ThrowIfEqual(num, 5)")]
+    [InlineData("ThrowIfGreaterThan(num, 6)")]
+    [InlineData("ThrowIfGreaterThanOrEqual(num, 1)")]
+    [InlineData("ThrowIfLessThan(num, 2)")]
+    [InlineData("ThrowIfLessThanOrEqual(num, 3)")]
+    [InlineData("ThrowIfEqual(num, 15)")]
+    [InlineData("ThrowIfNotEqual(num, 10)")]
+    [InlineData("ThrowIfZero(num)")]
```

Comment:
Although we have refactorrings only for `ThrowIfNegative` and `ThrowIfNegativeOrZero` as the most common ones we recognize other numeric checks as well since adding our own if one of this is present doesn't make much sense